Mathlib/FieldTheory/RatFunc/AsPolynomial.lean

```diff
@@ -207,6 +210,76 @@ theorem valuation_of_mk (f : Polynomial K) {g : Polynomial K} (hg : g ≠ 0) :
       (Polynomial.idealX K).intValuation f / (Polynomial.idealX K).intValuation g := by
   simp only [RatFunc.mk_eq_mk' _ hg, valuation_of_mk']
 
+end HeightOneSpectrum
+
+section TrivialOnConstants
+
+variable {Γ : Type*}
+
+section LinearOrderedCommMonoidWithZero
+
+variable [LinearOrderedCommMonoidWithZero Γ] {v : Valuation (RatFunc K) Γ}
+
+open Valuation
+
+/-- If a valuation `v` is trivial on constants then for every `n : ℕ` the valuation of
+`(monomial n a)` is equal to `(v RatFunc.X) ^ n`. -/
+lemma valuation_monomial_eq_valuation_X_pow (hv : ∀ a : K, a ≠ 0 → v (C a) = 1) (n : ℕ) {a : K}
+    (ha : a ≠ 0) : v (monomial n a) = v RatFunc.X ^ n := by
+  simp_all [RatFunc.coePolynomial, ← C_mul_X_pow_eq_monomial]
+
+/-- If a valuation `v` is trivial on constants and `1 < v RatFunc.X` then for every polynomial `p`,
+`v p = v RatFunc.X ^ p.natDegree`.
+
+Note: The condition `1 < v RatFunc.X` is typically satisfied by the valuation at infinity. -/
+theorem valuation_eq_valuation_X_pow_natDegree_of_one_lt_valuation_X [Nontrivial Γ]
+    [PosMulStrictMono Γ] (hlt : 1 < v RatFunc.X) (h : ∀ a : K, a ≠ 0 → v (C a) = 1) {p : K[X]}
+    (hp : p ≠ 0) : v p = v RatFunc.X ^ p.natDegree := by
+  replace h : ∀ a : K, a ≠ 0 → v (RatFunc.C a) = 1 := h
+  rw [RatFunc.coePolynomial,
+    ← valuation_monomial_eq_valuation_X_pow _ h _ (leadingCoeff_ne_zero.mpr hp)]
+  nth_rw 1 [as_sum_range p, map_sum]
+  apply map_sum_eq_of_lt _ (by simp) (by simp [hp])
+  intro i hi
+  simp only [Finset.mem_sdiff, Finset.mem_range, Finset.mem_singleton, Nat.lt_add_one_iff,
+    ← lt_iff_le_and_ne] at hi
+  simp_all only [ne_eq, RatFunc.algebraMap_eq_C, ← C_mul_X_pow_eq_monomial, _root_.map_mul,
+    RatFunc.algebraMap_C, _root_.map_pow, RatFunc.algebraMap_X, coeff_natDegree,
+    leadingCoeff_eq_zero, not_false_eq_true, one_mul, gt_iff_lt]
+  by_cases h0 : (p.coeff i) = 0
+  · simp only [h0, _root_.map_zero, zero_mul, one_mul]
+    exact pow_pos (lt_trans zero_lt_one hlt) p.natDegree
+  simp only [h _ h0, one_mul, pow_lt_pow_right₀ hlt hi]
+
+/-- If a valuation `v` is trivial on constants and `v RatFunc.X ≤ 1` then `for every polynomial `p`,
+`v p ≤ 1`. -/
+theorem valuation_le_one_of_valuation_X_le_one (h : ∀ a : K, a ≠ 0 → v (C a) = 1)
+    (hle : v RatFunc.X ≤ 1) (p : K[X]) : v p ≤ 1 := by
+  rw [as_sum_range p, RatFunc.coePolynomial, map_sum]
+  apply map_sum_le v
+  intro i _
+  by_cases h0 : p.coeff i = 0
+  · simp_all
+  · rw [← RatFunc.coePolynomial]
+    simp_all [valuation_monomial_eq_valuation_X_pow, pow_le_one']
+
+end LinearOrderedCommMonoidWithZero
+
+section LinearOrderedCommGroupWithZero
```

Comment:
I think that there is no need for this new section: since it contains a unique lemma, you can close the previous one and simply add the variable `[LinearOrderedCommGroupWithZero Γ]` to the lemma below.